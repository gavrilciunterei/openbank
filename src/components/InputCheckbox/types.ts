export type CheckboxProps = {
  checked: boolean;
  text: string;
  handleOnChange: React.ChangeEventHandler<HTMLInputElement>;
  name?: string;
};
