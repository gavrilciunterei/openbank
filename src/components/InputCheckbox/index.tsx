import React from 'react';
import { CheckboxProps } from './types';

function InputCheckbox({ checked, handleOnChange, text, name }: CheckboxProps) {
  return (
    <label>
      <input
        name={name}
        type="checkbox"
        checked={checked}
        onChange={handleOnChange}
      />
      {text}
    </label>
  );
}

export default InputCheckbox;
