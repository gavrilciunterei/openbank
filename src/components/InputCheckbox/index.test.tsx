import React from 'react';
import renderer from 'react-test-renderer';
import InputCheckbox from './index';

it('Test InputCheckbox', () => {
  const tree = renderer
    .create(
      <InputCheckbox
        checked={false}
        handleOnChange={() => true}
        text="Prueba checkbox"
        name="Check"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
