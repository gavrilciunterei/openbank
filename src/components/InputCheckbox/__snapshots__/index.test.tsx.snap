// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InputCheckbox 1`] = `
<label>
  <input
    checked={false}
    name="Check"
    onChange={[Function]}
    type="checkbox"
  />
  Prueba checkbox
</label>
`;
